auction/chaincode-go/smart-contract/utils.go

```diff
@@ -0,0 +1,107 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package auction
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric-chaincode-go/pkg/statebased"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// setAssetStateBasedEndorsement sets the endorsement policy of a new auction
+func setAssetStateBasedEndorsement(ctx contractapi.TransactionContextInterface, auctionID string, orgToEndorse string) error {
+
+	endorsementPolicy, err := statebased.NewStateEP(nil)
+	if err != nil {
+		return err
+	}
+	err = endorsementPolicy.AddOrgs(statebased.RoleTypePeer, orgToEndorse)
+	if err != nil {
+		return fmt.Errorf("failed to add org to endorsement policy: %v", err)
+	}
+	policy, err := endorsementPolicy.Policy()
+	if err != nil {
+		return fmt.Errorf("failed to create endorsement policy bytes from org: %v", err)
+	}
+	err = ctx.GetStub().SetStateValidationParameter(auctionID, policy)
+	if err != nil {
+		return fmt.Errorf("failed to set validation parameter on auction: %v", err)
+	}
+
+	return nil
+}
+
+// addAssetStateBasedEndorsement adds a new organization as an endorser of the auction
+func addAssetStateBasedEndorsement(ctx contractapi.TransactionContextInterface, auctionID string, orgToEndorse string) error {
+
+	endorsementPolicy, err := ctx.GetStub().GetStateValidationParameter(auctionID)
+	if err != nil {
+		return err
+	}
+
+	newEndorsementPolicy, err := statebased.NewStateEP(endorsementPolicy)
+	if err != nil {
+		return err
+	}
+
+	err = newEndorsementPolicy.AddOrgs(statebased.RoleTypePeer, orgToEndorse)
+	if err != nil {
+		return fmt.Errorf("failed to add org to endorsement policy: %v", err)
+	}
+	policy, err := newEndorsementPolicy.Policy()
+	if err != nil {
+		return fmt.Errorf("failed to create endorsement policy bytes from org: %v", err)
+	}
+	err = ctx.GetStub().SetStateValidationParameter(auctionID, policy)
+	if err != nil {
+		return fmt.Errorf("failed to set validation parameter on auction: %v", err)
+	}
+
+	return nil
+}
+
+// getCollectionName is an internal helper function to get collection of submitting client identity.
+func getCollectionName(ctx contractapi.TransactionContextInterface) (string, error) {
+
+	// Get the MSP ID of submitting client identity
+	clientMSPID, err := ctx.GetClientIdentity().GetMSPID()
+	if err != nil {
+		return "", fmt.Errorf("failed to get verified MSPID: %v", err)
+	}
+
+	// Create the collection name
+	orgCollection := "_implicit_org_" + clientMSPID
+
+	return orgCollection, nil
+}
+
+// verifyClientOrgMatchesPeerOrg is an internal function used verify client org id and matches peer org id.
```

Comment:
used to verify client org id matches peer org id.

Reply:
done